### PR TITLE
Create space

### PR DESCRIPTION
**Pull request by swizelc**

### CODE REVIEW

File: lib/space.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class Space
+
+  attr_reader :name, :description, :price, :date, :availability
+
+  def initialize(name:, description:, price:, date:, availability:)
+    @name = name
+    @description = description
+    @price = price
+    @date = date
+    @availability = availability
```
Comment (swizelc) [inline lib/space.rb:12]: availability should probably hold the dates in a list

File: lib/space.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+class Space
+
+  attr_reader :name, :description, :price, :date, :availability
+
+  def initialize(name:, description:, price:, date:, availability:)
+    @name = name
+    @description = description
+    @price = price
+    @date = date
```
Comment (swizelc) [inline lib/space.rb:11]: dates - more than one

File: lib/space.rb (hunk 1 above)
Comment (swizelc) [inline lib/space.rb:12]: booked or not booked - if booked take out the db

File: spec/features/featuretest_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    scenario 'User views all spaces' do
+        Space.create_space(name: 'The house', description: 'It is a big house', price: 134.67, date: '2020-10-2', availability: 'Available')
+        visit('/spaces')
+        expect(page).to have_content 'The house'
+    end
+end
+
+feature 'Creates a new space' do
+  scenario 'User creates a new space' do
+    visit('/create_space')
```
Comment (swizelc) [inline spec/features/featuretest_spec.rb:13]: make a web helper

File: spec/space_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require 'space'
+
+describe Space do
+  it 'responds to method .all' do
+    expect(Space).to respond_to(:all)
+  end
+
+  describe '.all' do
+    it 'list all spaces' do
+      connection = PG.connect(dbname: 'makersbnb_test')
```
Comment (swizelc) [inline spec/space_spec.rb:10]: remove sql from tests

File: spec/space_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      connection.exec("INSERT INTO spaces (name, description, price, date, availability) VALUES ('The house', 'This is a big house', 123.50, '2020-11-06', 'Available');")
+      spaces = Space.all
+      expect(spaces).to include 'The house'
+    end
+  end
+  #need to see all the arguments displayed
+
+  #describe '.create_space' do
```
Comment (swizelc) [inline spec/space_spec.rb:18]: do this test

File: views/all_spaces.erb [hunk 6]
```diff
@@ -0,0 +1 @@
+<%= @spaces %>
```
Comment (swizelc) [inline views/all_spaces.erb:1]: make it prettier making it list everything including not just names

**Review by christian-naykene — APPROVED** [review]

**Pull request merged**